Fix #49, refactor write functions to mesh

ezrayst opened the pull request.

Commit:
:art: refactor output files to mesh

Commit:
:art: refactor output files to mesh

Commit:
:art: refactor output files to mesh

Commit:
:art: refactor output files to mesh

cw646 commented on include/mesh.h:
```diff
@@ -73,6 +85,9 @@ class Mesh {
 
+  //! Map to store element id and volume of the element
+  std::map<unsigned, double> volumes_;
+
```
@kks32 didn't we decide not to store volumes?

ezrayst replied:
@cw646, I was just thinking it is cleaner and more consistent this way.

Previously you just pass the whole map to `io` and now that we are doing it in `mesh` I can make it not to store and to have just one function to `compute` and to `write`. What do you think?

kks32 replied:
Material points will know it's volume, can we just iterate over points and print volumes?

Commit:
:wrench: refactor volume writing

Commit:
:wrench: refactor volume writing

Commit:
:wrench: refactor volume writing

Commit:
:wrench: refactor volume writing

cw646 committed:
Fix Issue #51 and #52 (#53)

* :art: Refactor io, Issue #51
:art: Move command line parsing to io, Issue #52
:art: moved material properties file to src
:art: updates to cmake

* :wrench: fixed cmake for regression test

* :art: removed unused const and member variables

* :fire: Remove redundant initialisation

* :art: Fix comments

kks32 committed:
:fire: Remove travis-ci from develop (#56)

kks32 reviewed (changes requested):
Could you please rebase with the develop branch, as IO handling has changed.

Commit:
:paperclip: merge from develop

Commit:
:paperclip: merge from develop

Commit:
:paperclip: merge from develop

Commit:
:wrench: fix gauss points

cw646 committed:
:wrench: Fixes issue of points not generating (#57)

Commit:
:paperclip: merge to develop
